compel/arch/aarch64/src/lib/infect.c

```diff
@@ -297,3 +298,43 @@ int ptrace_flush_breakpoints(pid_t pid, bool restore)
 
 	return 0;
 }
+
+static bool task_is_trapped(int status, pid_t pid)
+{
+	if (WIFSTOPPED(status) && WSTOPSIG(status) == SIGTRAP)
+		return true;
+
+	pr_err("Task %d is in unexpected state: %x\n", pid, status);
+	if (WIFEXITED(status))
+		pr_err("Task exited with %d\n", WEXITSTATUS(status));
+	if (WIFSIGNALED(status))
+		pr_err("Task signaled with %d: %s\n", WTERMSIG(status), strsignal(WTERMSIG(status)));
+	if (WIFSTOPPED(status))
+		pr_err("Task stopped with %d: %s\n", WSTOPSIG(status), strsignal(WSTOPSIG(status)));
+	if (WIFCONTINUED(status))
+		pr_err("Task continued\n");
+
+	return false;
+}
+
+/* wait single step debug trap event */
+int arch_wait_tasks_trap(int tasks)
```

Comment:
arch_wait_tasks_trap has to be removed. You have it only because breakpoints don't work properly right now.